### ISSUE TITLE
It's too easy to create a live service where a Fake was intended

### ISSUE THREAD

**Issue by blairconrad** [blairconrad/SelfInitializingFakes#37]
Consider:

```c#
SelfInitializingFake.For(() => new WebComicFetcher(), callRepository);
```

Where `WebComicFetcher` has no virtual methods. This will succeed, but the created fake will not be able to intercept any methods. The service will be "live".

Instead, we should differentiate between the faked service type and the type of the real service that's used when there's no call repository. This will require an API change:

```c#
SelfInitializingFake.For<TService>(
    Func<TService> factory,
    IRecordedCallRepository callRepository)
```

will become

```c#
SelfInitializingFake<TService>.For<TConcreteServiceService>(
    Func<TConcreteService> factory,
    IRecordedCallRepository callRepository)
```

**Comment by thomaslevesque** [comment]
👍

**Comment by blairconrad** [comment]
This change was released as part of [0.2.0](https://github.com/blairconrad/SelfInitializingFakes/releases/tag/0.2.0).
